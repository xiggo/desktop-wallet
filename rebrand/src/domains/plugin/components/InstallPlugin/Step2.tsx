import { prettyBytes } from "@payvo/sdk-helpers";
import React from "react";
import { useTranslation } from "react-i18next";

import { CircularProgressBar } from "@/app/components/CircularProgressBar";
import { PluginImage } from "@/domains/plugin/components/PluginImage";
import { ExtendedSerializedPluginConfigurationData, SerializedPluginConfigurationData } from "@/plugins";

interface Properties {
	plugin: SerializedPluginConfigurationData | ExtendedSerializedPluginConfigurationData;
	downloadProgress: { percent?: number; transferredBytes?: number; totalBytes: number };
}

export const SecondStep = ({ plugin, downloadProgress }: Properties) => {
	const { t } = useTranslation();

	const hasSize = plugin.size !== "0 B";

	const percent = Math.floor((downloadProgress.percent || 0) * 100);

	return (
		<section data-testid="InstallPlugin__step--second">
			<div className="flex mt-8">
				<PluginImage className="mr-6" size="lg" logoURL={plugin.logo} />

				<div className="flex-1">
					<div className="flex flex-col justify-around h-full">
						<div>
							<p className="text-sm font-semibold text-theme-secondary-400">{t("COMMON.PLUGIN")}</p>
							<p className="text-lg font-semibold text-theme-black">{plugin.title}</p>
						</div>
						<div className="flex justify-between">
							<span>
								<p className="text-sm font-semibold text-theme-secondary-400">
									{t("COMMON.DOWNLOADING")}...
								</p>
								<p
									data-testid="InstallPlugin__step--second__progress"
									className="font-bold text-theme-secondary-text"
								>
									{prettyBytes(downloadProgress.transferredBytes ?? 0)} /{" "}
									{hasSize ? plugin.size : prettyBytes(downloadProgress.totalBytes)}
								</p>
							</span>
							<div className="mr-2">
								<CircularProgressBar value={percent} size={50} strokeWidth={3} fontSize={0.8} />
							</div>
						</div>
					</div>
				</div>
			</div>
		</section>
	);
};
