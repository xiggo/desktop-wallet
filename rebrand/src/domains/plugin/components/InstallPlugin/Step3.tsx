import React from "react";
import { useTranslation } from "react-i18next";

import { Circle } from "@/app/components/Circle";
import { Icon } from "@/app/components/Icon";
import { PluginImage } from "@/domains/plugin/components/PluginImage";

interface Properties {
	plugin: any;
}

export const ThirdStep = ({ plugin }: Properties) => {
	const { t } = useTranslation();

	return (
		<section data-testid="InstallPlugin__step--third">
			<p className="-mt-1 text-theme-secondary-text">{t("PLUGINS.MODAL_ENABLE_PLUGIN.DESCRIPTION")}</p>

			<div className="flex mt-8">
				<PluginImage className="mr-6" size="lg" logoURL={plugin.logo} />

				<div className="flex-1">
					<div className="flex flex-col justify-around h-full">
						<div>
							<p className="text-sm font-semibold text-theme-secondary-400">{t("COMMON.PLUGIN")}</p>
							<p className="text-lg font-semibold text-theme-black">{plugin.title}</p>
						</div>
						<div className="flex justify-between">
							<span>
								<p className="text-sm font-semibold text-theme-secondary-400">
									{t("PLUGINS.MODAL_ENABLE_PLUGIN.INSTALLATION")}
								</p>
								<p className="font-bold text-theme-secondary-text">{t("COMMON.COMPLETED")}</p>
							</span>
							<div className="">
								<Circle
									size="lg"
									className="relative z-10 bg-theme-background border-theme-secondary-300 dark:border-theme-secondary-800"
								>
									<span className="text-theme-success-600">
										<Icon name="Checkmark" />
									</span>
								</Circle>
								<Circle size="lg" className="relative z-0 bg-theme-background border-theme-success-600">
									<span className="text-xs font-semibold text-theme-success-600">100%</span>
								</Circle>
							</div>
						</div>
					</div>
				</div>
			</div>
		</section>
	);
};
