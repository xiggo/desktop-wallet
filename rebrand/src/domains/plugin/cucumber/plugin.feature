Feature: Plugins

    @plugins-filterGame
    Scenario: Filter plugins by Game category
        Given Alice is on the plugins page
        When she filters plugins by game category
        Then only game plugins are displayed

    @plugins-filterUtility
    Scenario: Filter plugins by Utility category
        Given Alice is on the plugins page
        When she filters plugins by utility category
        Then only utility plugins are displayed

    @plugins-filterOther
    Scenario: Filter plugins by Other category
        Given Alice is on the plugins page
        When she filters plugins by other category
        Then only other plugins are displayed

    @plugins-filterMyPlugins
    Scenario: Filter plugins by My Plugins
        Given Alice is on the plugins page
        When she filters plugins by selecting my plugins
        Then only installed plugins are displayed
    
    @plugins-navigateToPluginDetails
    Scenario: Navigate to Plugin Details page and back
        Given Alice is on the plugins page
        When she clicks on a plugin
        Then she is navigated to the plugin details page
        When she selects plugins from the navbar
        Then she is navigated back to the plugin page 
