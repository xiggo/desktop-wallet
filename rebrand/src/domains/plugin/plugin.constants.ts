export const pluginData = {
	about: "Use the ARK Explorer to get full visibility of critical data from the ARK network. Data such as the latest blocks, wallet addresses and transactions. Plus monitor delegate status, their position and more.",
	author: "Payvo",
	averageRating: "4.6",
	category: "Utility",
	name: "ARK Explorer",
	permissions: ["Embedded Webpages", "API Requests", "Access to Profiles"],
	screenshots: [1, 2, 3],
	size: "4.2",
	url: "github.com",
	version: "1.3.8",
} as const;

export const plugins = [
	{
		author: "Payvo",
		category: "Utility",
		description: "This is a description",
		id: 0,
		isOfficial: true,
		name: "ARK Explorer",
		rating: 4.6,
		version: "1.3.8",
	},
	{
		author: "Breno Polanski",
		category: "Utility",
		description: "This is a description",
		id: 1,
		name: "Animal Avatars",
		rating: 4.6,
		version: "1.3.8",
	},
	{
		author: "ChangeNOW",
		category: "Other",
		description: "This is a description",
		id: 2,
		name: "ChangeNOW Plugin",
		rating: 4.8,
		version: "1.3.8",
	},
	{
		author: "Delegate Fun",
		category: "Game",
		description: "This is a description",
		id: 4,
		name: "Bold Ninja",
		rating: 4.9,
		version: "2.0.0",
	},
] as const;

export const PLUGIN_CATEGORIES = ["gaming", "utility", "other"] as const;
