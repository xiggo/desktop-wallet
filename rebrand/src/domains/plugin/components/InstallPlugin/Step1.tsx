import React from "react";
import { useTranslation } from "react-i18next";

import { PluginPermissions } from "@/domains/plugin/plugin.contracts";

interface Properties {
	plugin: any;
}

export const FirstStep = ({ plugin }: Properties) => {
	const { t, i18n } = useTranslation();
	const translatedPermissions = (plugin.permissions as PluginPermissions[]).map((permission) => {
		const key = `PLUGINS.PERMISSIONS.${permission}` as const;
		return i18n.exists(key) ? t(key) : permission;
	});

	return (
		<section data-testid="InstallPlugin__step--first">
			<p className="mt-7 text-lg font-semibold text-theme-secondary-text">
				{t("PLUGINS.MODAL_INSTALL_PLUGIN.DESCRIPTION")}
			</p>
			<div className="max-w-sm">
				<ul className="mt-2 ml-5 leading-8 list-outside list-disc text-theme-secondary-text">
					{translatedPermissions.map((permission: string) => (
						<li key={permission}>{permission}</li>
					))}
				</ul>
			</div>
		</section>
	);
};
