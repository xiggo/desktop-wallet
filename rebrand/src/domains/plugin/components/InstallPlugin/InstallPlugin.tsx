import { Contracts } from "@payvo/sdk-profiles";
import { ipcRenderer } from "electron";
import { ExtendedSerializedPluginConfigurationData, SerializedPluginConfigurationData } from "plugins";
import React, { useCallback, useEffect, useState } from "react";
import { useTranslation } from "react-i18next";
import tw, { styled } from "twin.macro";

import { FirstStep } from "./Step1";
import { SecondStep } from "./Step2";
import { ThirdStep } from "./Step3";
import { Button } from "@/app/components/Button";
import { Modal } from "@/app/components/Modal";
import { TabPanel, Tabs } from "@/app/components/Tabs";
import { toasts } from "@/app/services";
import { usePluginManagerContext } from "@/plugins/context/PluginManagerProvider";
import { assertString } from "@/utils/assertions";

interface InstallPluginProperties {
	isOpen: boolean;
	onClose?: () => void;
	onCancel?: () => void;
	plugin: SerializedPluginConfigurationData | ExtendedSerializedPluginConfigurationData;
	repositoryURL?: string;
	profile: Contracts.IProfile;
}

enum Step {
	PermissionsStep = 1,
	DownloadStep,
	EnableStep,
}

const ButtonWrapper = styled.div`
	${tw`flex justify-end mt-8 space-x-3`}
`;

export const InstallPlugin = ({
	isOpen,
	onClose,
	onCancel,
	plugin,
	profile,
	repositoryURL,
}: InstallPluginProperties) => {
	const { t } = useTranslation();
	const { downloadPlugin, installPlugin, trigger, pluginManager } = usePluginManagerContext();
	const [activeStep, setActiveStep] = useState<Step>(Step.PermissionsStep);
	const [downloadProgress, setDownloadProgress] = useState<any>({ totalBytes: 0 });
	const [downloadResult, setDownloadResult] = useState<any>();
	const profileId = profile.id();

	const handleInstall = useCallback(
		async (result: any) => {
			try {
				await installPlugin(result?.savedPath, plugin.id, profileId, result?.subDirectory);
				setActiveStep(Step.EnableStep);
			} catch (error) {
				/* istanbul ignore next */
				toasts.error(
					t("PLUGINS.MODAL_INSTALL_PLUGIN.INSTALL_FAILURE", { msg: error.message, name: plugin.title }),
				);
				onClose?.();
			}
		},
		[installPlugin, plugin, profileId, t, onClose],
	);

	const handleDownload = useCallback(async () => {
		setActiveStep(Step.DownloadStep);
		let result: any;
		let downloadFailure = false;

		try {
			result = await downloadPlugin(plugin, repositoryURL);
			setDownloadResult(result);
		} catch {
			downloadFailure = true;
			toasts.error(t("PLUGINS.MODAL_INSTALL_PLUGIN.DOWNLOAD_FAILURE", { name: plugin.title }));
			onClose?.();
		}

		if (!downloadFailure) {
			assertString(result?.savedPath);

			await handleInstall(result);
		}
	}, [handleInstall, downloadPlugin, plugin, repositoryURL, t, onClose]);

	const handleEnable = useCallback(() => {
		try {
			pluginManager.plugins().findById(plugin.id)?.enable(profile, { autoRun: true });
			trigger();

			toasts.success(t("PLUGINS.ENABLE_SUCCESS", { name: plugin.title }));
		} catch (error) {
			toasts.error(t("PLUGINS.ENABLE_FAILURE", { msg: error.message, name: plugin.title }));
		} finally {
			onClose?.();
		}
	}, [pluginManager, plugin, profile, trigger, t, onClose]);

	const onModalClose = () => {
		if (activeStep === Step.EnableStep) {
			ipcRenderer.invoke("plugin:cancel-install", { savedPath: downloadResult.savedPath });
			setActiveStep(Step.PermissionsStep);
			setDownloadResult(undefined);
		}

		onClose?.();
	};

	useEffect(() => {
		// @ts-ignore
		const listener = (_, value: any) => setDownloadProgress(value);
		ipcRenderer.on("plugin:download-progress", listener);

		return () => {
			ipcRenderer.removeListener("plugin:download-progress", listener);
		};
	}, []);

	const renderModalTitle = () =>
		({
			[Step.PermissionsStep]: () => t("COMMON.PERMISSIONS"),
			[Step.DownloadStep]: () => t("COMMON.DOWNLOADING"),
			[Step.EnableStep]: () => t("PLUGINS.SUCCESSFULLY_INSTALLED"),
		}[activeStep]());

	return (
		<Modal title={renderModalTitle()} size="lg" isOpen={isOpen} onClose={onModalClose}>
			<Tabs activeId={activeStep}>
				<TabPanel tabId={Step.PermissionsStep}>
					<FirstStep plugin={plugin} />
				</TabPanel>

				<TabPanel tabId={Step.DownloadStep}>
					<SecondStep plugin={plugin} downloadProgress={downloadProgress} />
				</TabPanel>

				<TabPanel tabId={Step.EnableStep}>
					<ThirdStep plugin={plugin} />
				</TabPanel>

				{activeStep === Step.PermissionsStep && (
					<ButtonWrapper>
						<Button variant="secondary" onClick={onCancel} data-testid="InstallPlugin__cancel-button">
							{t("COMMON.CANCEL")}
						</Button>
						<Button onClick={handleDownload} data-testid="InstallPlugin__allow-button">
							{t("PLUGINS.MODAL_INSTALL_PLUGIN.ALLOW")}
						</Button>
					</ButtonWrapper>
				)}

				{activeStep === Step.EnableStep && (
					<ButtonWrapper>
						<Button
							variant="secondary"
							onClick={() => {
								ipcRenderer.invoke("plugin:cancel-install", { savedPath: downloadResult.savedPath });
								setActiveStep(Step.PermissionsStep);
								setDownloadResult(undefined);

								onCancel?.();
							}}
							data-testid="InstallPlugin__cancel-button"
						>
							{t("COMMON.CANCEL")}
						</Button>

						<Button onClick={handleEnable} data-testid="InstallPlugin__enable-button">
							{t("COMMON.ENABLE")}
						</Button>
					</ButtonWrapper>
				)}
			</Tabs>
		</Modal>
	);
};
