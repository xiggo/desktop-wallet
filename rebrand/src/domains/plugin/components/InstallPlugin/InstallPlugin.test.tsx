import { Contracts } from "@payvo/sdk-profiles";
import userEvent from "@testing-library/user-event";
import { ipcRenderer } from "electron";
import { PluginManager } from "plugins";
import React from "react";

import { InstallPlugin } from "./InstallPlugin";
import { FirstStep } from "./Step1";
import { SecondStep } from "./Step2";
import { ThirdStep } from "./Step3";
import { toasts } from "@/app/services";
import { translations as pluginTranslations } from "@/domains/plugin/i18n";
import { PluginManagerProvider } from "@/plugins/context/PluginManagerProvider";
import { env, getDefaultProfileId, pluginManager, render, screen, waitFor } from "@/utils/testing-library";

const ipcRendererMockImplementation = (channel: string, listener: (...arguments_: any[]) => void) => {
	if (channel === "plugin:download-progress") {
		return listener(undefined, { totalBytes: 200 });
	}
};

describe("InstallPlugin", () => {
	let profile: Contracts.IProfile;

	beforeEach(() => {
		profile = env.profiles().findById(getDefaultProfileId());
	});

	it("should not render if not open", () => {
		const { asFragment } = render(
			<PluginManagerProvider manager={new PluginManager()} services={[]}>
				<InstallPlugin profile={profile} isOpen={false} />
			</PluginManagerProvider>,
		);

		expect(screen.queryByTestId("modal__inner")).not.toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render 1st step", () => {
		const { asFragment } = render(
			<FirstStep plugin={{ permissions: ["PROFILE", "EVENTS", "HTTP", "CUSTOM_PERMISSION"] }} />,
		);

		expect(screen.getByTestId("InstallPlugin__step--first")).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render 2nd step with partial download progress", () => {
		const { asFragment } = render(
			<SecondStep plugin={{ size: "0 B", title: "My Plugin" }} downloadProgress={{ totalBytes: 100 }} />,
		);

		expect(screen.getByTestId("InstallPlugin__step--second__progress")).toHaveTextContent("0 B / 100 B");
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render 2nd step with full download progress", () => {
		const { asFragment } = render(
			<SecondStep
				plugin={{ logo: "https://payvo.com/logo.png", size: "100 B", title: "My Plugin" }}
				downloadProgress={{ percent: 0.843_471_432_579, totalBytes: 100, transferredBytes: 84.347_143_257_9 }}
			/>,
		);

		expect(screen.getByTestId("CircularProgressBar__percentage")).toHaveTextContent("84%");
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render 3rd step", () => {
		const { asFragment } = render(<ThirdStep plugin={{ title: "My Plugin" }} />);

		expect(screen.getByTestId("InstallPlugin__step--third")).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render 3rd step with plugin logo", () => {
		const { asFragment } = render(
			<ThirdStep plugin={{ logo: "https://payvo.com/logo.png", title: "My Plugin" }} />,
		);

		expect(screen.getByTestId("PluginImage")).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should download, install and enable plugin", async () => {
		const toastSpy = jest.spyOn(toasts, "success").mockImplementation();

		const cancelInstall = jest.fn();

		const plugin = {
			id: "remote-plugin",
			permissions: ["PROFILE"],
			title: "Remote Plugin",
		};

		jest.spyOn(ipcRenderer, "on").mockImplementation(ipcRendererMockImplementation as any);

		const invokeSpy = jest.spyOn(ipcRenderer, "invoke").mockImplementation((channel, ...arguments_) => {
			if (channel === "plugin:loader-fs.find") {
				return Promise.resolve({
					config: { keywords: ["@payvo", "wallet-plugin"], name: plugin.id, version: "0.0.1" },
					dir: `/plugins/${plugin.id}`,
					source: "module.exports = () => void 0",
					sourcePath: `/plugins/${plugin.id}/index.js`,
				});
			}

			if (channel === "plugin:install") {
				return Promise.resolve(`/Users/plugins/${plugin.id}`);
			}

			if (channel === "plugin:download") {
				return Promise.resolve(`/Users/plugins/temp/${plugin.id}`);
			}

			if (channel === "plugin:cancel-install") {
				cancelInstall(...arguments_);
			}

			return Promise.resolve();
		});
		const onClose = jest.fn();

		render(
			<PluginManagerProvider manager={pluginManager} services={[]}>
				<InstallPlugin
					profile={profile}
					repositoryURL="https://github.com/author/plugin"
					isOpen={true}
					plugin={plugin}
					onClose={onClose}
				/>
			</PluginManagerProvider>,
		);

		userEvent.click(screen.getByTestId("modal__close-btn"));

		expect(onClose).toHaveBeenCalledWith();

		const downloadAndInstallPlugin = async () => {
			userEvent.click(screen.getByTestId("InstallPlugin__allow-button"));

			expect(screen.getByTestId("InstallPlugin__step--second")).toBeVisible();

			await waitFor(() => {
				expect(invokeSpy).toHaveBeenCalledWith("plugin:download", {
					name: plugin.id,
					url: "https://github.com/author/plugin/archive/master.zip",
				});
			});

			expect(invokeSpy).toHaveBeenCalledWith("plugin:install", {
				name: plugin.id,
				profileId: profile.id(),
				savedPath: `/Users/plugins/temp/${plugin.id}`,
			});

			expect(invokeSpy).toHaveBeenLastCalledWith("plugin:loader-fs.find", `/Users/plugins/${plugin.id}`);

			await expect(screen.findByTestId("InstallPlugin__step--third")).resolves.toBeVisible();
		};

		await downloadAndInstallPlugin();

		// Cancel: do not go ahead with enable

		userEvent.click(screen.getByTestId("InstallPlugin__cancel-button"));

		const cancelPluginCalled = () =>
			expect(cancelInstall).toHaveBeenCalledWith({
				savedPath: `/Users/plugins/temp/${plugin.id}`,
			});
		await waitFor(cancelPluginCalled);

		expect(cancelInstall).toHaveBeenCalledTimes(1);

		await downloadAndInstallPlugin();

		// Close: do not go ahead with enable

		userEvent.click(screen.getByTestId("modal__close-btn"));

		await waitFor(cancelPluginCalled);

		expect(cancelInstall).toHaveBeenCalledTimes(2);

		// Happy path: proceed with enable

		await downloadAndInstallPlugin();

		expect(screen.getByText(pluginTranslations.MODAL_ENABLE_PLUGIN.DESCRIPTION)).toBeInTheDocument();

		userEvent.click(screen.getByTestId("InstallPlugin__enable-button"));

		expect(onClose).toHaveBeenCalledTimes(3);

		expect(pluginManager.plugins().findById(plugin.id)?.isEnabled(profile)).toBeTruthy();

		expect(toastSpy).toHaveBeenCalledWith(pluginTranslations.ENABLE_SUCCESS.replace("{{name}}", plugin.title));

		pluginManager.plugins().removeById(plugin.id, profile);

		toastSpy.mockRestore();
	});

	it("should fail to enable plugin", async () => {
		const toastSpy = jest.spyOn(toasts, "error").mockImplementation();

		const plugin = {
			id: "remote-plugin",
			permissions: ["PROFILE"],
			title: "Remote Plugin",
		};

		jest.spyOn(ipcRenderer, "on").mockImplementation(ipcRendererMockImplementation as any);

		jest.spyOn(ipcRenderer, "invoke").mockImplementation((channel) => {
			if (channel === "plugin:loader-fs.find") {
				return Promise.resolve({
					config: { keywords: ["@payvo", "wallet-plugin"], name: plugin.id, version: "0.0.1" },
					dir: `/plugins/${plugin.id}`,
					source: "{ incompatible: true }",
					sourcePath: `/plugins/${plugin.id}/index.js`,
				});
			}

			if (channel === "plugin:install") {
				return Promise.resolve(`/Users/plugins/${plugin.id}`);
			}

			if (channel === "plugin:download") {
				return Promise.resolve(`/Users/plugins/temp/${plugin.id}`);
			}

			return Promise.resolve();
		});

		render(
			<PluginManagerProvider manager={pluginManager} services={[]}>
				<InstallPlugin
					profile={profile}
					repositoryURL="https://github.com/author/plugin"
					isOpen={true}
					plugin={plugin}
				/>
			</PluginManagerProvider>,
		);

		userEvent.click(screen.getByTestId("InstallPlugin__allow-button"));

		expect(screen.getByTestId("InstallPlugin__step--second")).toBeVisible();

		await expect(screen.findByTestId("InstallPlugin__step--third")).resolves.toBeVisible();

		userEvent.click(screen.getByTestId("InstallPlugin__enable-button"));

		expect(pluginManager.plugins().findById(plugin.id)?.isEnabled(profile)).toBeFalsy();

		expect(toastSpy).toHaveBeenCalledWith(expect.stringMatching(/Failed to enable plugin/));

		pluginManager.plugins().removeById(plugin.id, profile);

		toastSpy.mockRestore();
	});
});
