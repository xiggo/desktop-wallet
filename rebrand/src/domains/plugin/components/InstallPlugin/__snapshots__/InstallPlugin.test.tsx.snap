// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstallPlugin should not render if not open 1`] = `<DocumentFragment />`;

exports[`InstallPlugin should render 1st step 1`] = `
<DocumentFragment>
  <section
    data-testid="InstallPlugin__step--first"
  >
    <p
      class="mt-7 text-lg font-semibold text-theme-secondary-text"
    >
      This plugin needs the following permissions:
    </p>
    <div
      class="max-w-sm"
    >
      <ul
        class="mt-2 ml-5 leading-8 list-outside list-disc text-theme-secondary-text"
      >
        <li>
          Allows access to the currently active profile
        </li>
        <li>
          Allows access to the Payvo Wallet events
        </li>
        <li>
          Allows performing external web requests
        </li>
        <li>
          CUSTOM_PERMISSION
        </li>
      </ul>
    </div>
  </section>
</DocumentFragment>
`;

exports[`InstallPlugin should render 2nd step with full download progress 1`] = `
<DocumentFragment>
  <section
    data-testid="InstallPlugin__step--second"
  >
    <div
      class="flex mt-8"
    >
      <div
        class="PluginImage__PluginImageWrapper-sc-10kztrf-0 cvQNdU bg-theme-primary-100 text-theme-primary-600 dark:bg-theme-secondary-800 dark:text-theme-secondary-700 mr-6"
        data-testid="PluginImage"
      >
        <img
          alt="Logo"
          class="object-cover w-full h-full"
          src="https://payvo.com/logo.png"
        />
      </div>
      <div
        class="flex-1"
      >
        <div
          class="flex flex-col justify-around h-full"
        >
          <div>
            <p
              class="text-sm font-semibold text-theme-secondary-400"
            >
              Plugin
            </p>
            <p
              class="text-lg font-semibold text-theme-black"
            >
              My Plugin
            </p>
          </div>
          <div
            class="flex justify-between"
          >
            <span>
              <p
                class="text-sm font-semibold text-theme-secondary-400"
              >
                Downloading...
              </p>
              <p
                class="font-bold text-theme-secondary-text"
                data-testid="InstallPlugin__step--second__progress"
              >
                84.3 B / 100 B
              </p>
            </span>
            <div
              class="mr-2"
            >
              <svg
                height="50"
                viewBox="0 0 50 50"
                width="50"
              >
                <circle
                  cx="25"
                  cy="25"
                  fill="transparent"
                  r="23.5"
                  stroke="var(--theme-color-success-200)"
                  stroke-width="3"
                />
                <circle
                  cx="25"
                  cy="25"
                  fill="transparent"
                  r="23.5"
                  stroke="var(--theme-color-success-600)"
                  stroke-dasharray="147.6548547187203"
                  stroke-dashoffset="23.624776754995253"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="3"
                  transform="rotate(-90 25 25)"
                />
                <text
                  data-testid="CircularProgressBar__percentage"
                  dy="0.3em"
                  fill="var(--theme-color-success-600)"
                  font-size="0.8rem"
                  font-weight="600"
                  text-anchor="middle"
                  x="50%"
                  y="50%"
                >
                  <tspan>
                    84
                    <tspan
                      dy="-0.6em"
                      font-size="0.4rem"
                      font-weight="bold"
                    >
                      %
                    </tspan>
                  </tspan>
                </text>
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`InstallPlugin should render 2nd step with partial download progress 1`] = `
<DocumentFragment>
  <section
    data-testid="InstallPlugin__step--second"
  >
    <div
      class="flex mt-8"
    >
      <div
        class="PluginImage__PluginImageWrapper-sc-10kztrf-0 cvQNdU bg-theme-primary-100 text-theme-primary-600 dark:bg-theme-secondary-800 dark:text-theme-secondary-700 mr-6"
        data-testid="PluginImage"
      >
        <svg>
          plugin-logo-placeholder.svg
        </svg>
      </div>
      <div
        class="flex-1"
      >
        <div
          class="flex flex-col justify-around h-full"
        >
          <div>
            <p
              class="text-sm font-semibold text-theme-secondary-400"
            >
              Plugin
            </p>
            <p
              class="text-lg font-semibold text-theme-black"
            >
              My Plugin
            </p>
          </div>
          <div
            class="flex justify-between"
          >
            <span>
              <p
                class="text-sm font-semibold text-theme-secondary-400"
              >
                Downloading...
              </p>
              <p
                class="font-bold text-theme-secondary-text"
                data-testid="InstallPlugin__step--second__progress"
              >
                0 B / 100 B
              </p>
            </span>
            <div
              class="mr-2"
            >
              <svg
                height="50"
                viewBox="0 0 50 50"
                width="50"
              >
                <circle
                  cx="25"
                  cy="25"
                  fill="transparent"
                  r="23.5"
                  stroke="var(--theme-color-success-200)"
                  stroke-width="3"
                />
                <circle
                  cx="25"
                  cy="25"
                  fill="transparent"
                  r="23.5"
                  stroke="var(--theme-color-success-600)"
                  stroke-dasharray="147.6548547187203"
                  stroke-dashoffset="147.6548547187203"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="3"
                  transform="rotate(-90 25 25)"
                />
                <text
                  data-testid="CircularProgressBar__percentage"
                  dy="0.3em"
                  fill="var(--theme-color-success-600)"
                  font-size="0.8rem"
                  font-weight="600"
                  text-anchor="middle"
                  x="50%"
                  y="50%"
                >
                  <tspan>
                    0
                    <tspan
                      dy="-0.6em"
                      font-size="0.4rem"
                      font-weight="bold"
                    >
                      %
                    </tspan>
                  </tspan>
                </text>
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`InstallPlugin should render 3rd step 1`] = `
<DocumentFragment>
  <section
    data-testid="InstallPlugin__step--third"
  >
    <p
      class="-mt-1 text-theme-secondary-text"
    >
      Would you like to enable the Plugin?
    </p>
    <div
      class="flex mt-8"
    >
      <div
        class="PluginImage__PluginImageWrapper-sc-10kztrf-0 cvQNdU bg-theme-primary-100 text-theme-primary-600 dark:bg-theme-secondary-800 dark:text-theme-secondary-700 mr-6"
        data-testid="PluginImage"
      >
        <svg>
          plugin-logo-placeholder.svg
        </svg>
      </div>
      <div
        class="flex-1"
      >
        <div
          class="flex flex-col justify-around h-full"
        >
          <div>
            <p
              class="text-sm font-semibold text-theme-secondary-400"
            >
              Plugin
            </p>
            <p
              class="text-lg font-semibold text-theme-black"
            >
              My Plugin
            </p>
          </div>
          <div
            class="flex justify-between"
          >
            <span>
              <p
                class="text-sm font-semibold text-theme-secondary-400"
              >
                Installation
              </p>
              <p
                class="font-bold text-theme-secondary-text"
              >
                Completed
              </p>
            </span>
            <div
              class=""
            >
              <div
                class="Circle__CircleWrapper-sc-d5u41q-0 egSsGQ relative z-10 bg-theme-background border-theme-secondary-300 dark:border-theme-secondary-800 ring-theme-background"
              >
                <span
                  class="text-theme-success-600"
                >
                  <div
                    class="sc-bdvvtL fCrhkg"
                    height="16"
                    width="16"
                  >
                    <svg>
                      check-mark.svg
                    </svg>
                  </div>
                </span>
              </div>
              <div
                class="Circle__CircleWrapper-sc-d5u41q-0 egSsGQ relative z-0 bg-theme-background border-theme-success-600 ring-theme-background"
              >
                <span
                  class="text-xs font-semibold text-theme-success-600"
                >
                  100%
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`InstallPlugin should render 3rd step with plugin logo 1`] = `
<DocumentFragment>
  <section
    data-testid="InstallPlugin__step--third"
  >
    <p
      class="-mt-1 text-theme-secondary-text"
    >
      Would you like to enable the Plugin?
    </p>
    <div
      class="flex mt-8"
    >
      <div
        class="PluginImage__PluginImageWrapper-sc-10kztrf-0 cvQNdU bg-theme-primary-100 text-theme-primary-600 dark:bg-theme-secondary-800 dark:text-theme-secondary-700 mr-6"
        data-testid="PluginImage"
      >
        <img
          alt="Logo"
          class="object-cover w-full h-full"
          src="https://payvo.com/logo.png"
        />
      </div>
      <div
        class="flex-1"
      >
        <div
          class="flex flex-col justify-around h-full"
        >
          <div>
            <p
              class="text-sm font-semibold text-theme-secondary-400"
            >
              Plugin
            </p>
            <p
              class="text-lg font-semibold text-theme-black"
            >
              My Plugin
            </p>
          </div>
          <div
            class="flex justify-between"
          >
            <span>
              <p
                class="text-sm font-semibold text-theme-secondary-400"
              >
                Installation
              </p>
              <p
                class="font-bold text-theme-secondary-text"
              >
                Completed
              </p>
            </span>
            <div
              class=""
            >
              <div
                class="Circle__CircleWrapper-sc-d5u41q-0 egSsGQ relative z-10 bg-theme-background border-theme-secondary-300 dark:border-theme-secondary-800 ring-theme-background"
              >
                <span
                  class="text-theme-success-600"
                >
                  <div
                    class="sc-bdvvtL fCrhkg"
                    height="16"
                    width="16"
                  >
                    <svg>
                      check-mark.svg
                    </svg>
                  </div>
                </span>
              </div>
              <div
                class="Circle__CircleWrapper-sc-d5u41q-0 egSsGQ relative z-0 bg-theme-background border-theme-success-600 ring-theme-background"
              >
                <span
                  class="text-xs font-semibold text-theme-success-600"
                >
                  100%
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
